app/vue/src/client/preview/v3/prepare.ts

```diff
@@ -0,0 +1,34 @@
+// @ts-ignore
+import { defineComponent, h } from 'vue';
+import type { ComponentOptions, Component } from 'vue3';
+import { StoryFnVueReturnType } from '../types';
+import { propsContainer } from './render';
+import { extractProps } from '../util';
+
+export function prepare(
+  rawStory: StoryFnVueReturnType,
+  innerStory?: ComponentOptions
+): Component | null {
+  let story: ComponentOptions;
+
+  if (typeof rawStory === 'string') {
+    story = { template: rawStory } as ComponentOptions;
+  } else if (rawStory != null) {
+    story = rawStory as ComponentOptions;
+  } else {
+    return null;
+  }
+
+  if (innerStory) {
+    story.components = { story: innerStory };
+    return story;
+  }
+
+  propsContainer.props = extractProps(story.props);
+
+  return defineComponent({
+    render() {
+      return h(story, propsContainer.props);
+    },
+  });
```

Comment:
I suspect some features don't work well due to the lack of `STORYBOOK_WRAPS` and `STORYBOOK_VALUES`.

- Inline rendering in Docs Addon uses both https://github.com/storybookjs/storybook/blob/34a2d2e13ac330f441b50248bae7d7ae41c2d87a/addons/docs/src/frameworks/vue/prepareForInline.ts#L8-L10
  - Probably this https://github.com/andoshin11/storybook-vue3/issues/1
- Dynamic Source Rendering in Docs Addon uses `STORYBOOK_WRAPS` to skip decorators https://github.com/storybookjs/storybook/blob/b747c5c2396786682f36e5ac9b78ea990b6f952d/addons/docs/src/frameworks/vue/sourceDecorator.ts#L191-L205
- Storyshots Addon uses `STORYBOOK_WRAPS` for some reason https://github.com/storybookjs/storybook/blob/736f50e8e7f115a7bae9d90f05c861833428dd61/addons/storyshots/storyshots-core/src/frameworks/vue/renderTree.ts#L5

I'm not sure we can still inject these properties though 🤔

Reply:
Thanks @pocka , I'll check these ASAP